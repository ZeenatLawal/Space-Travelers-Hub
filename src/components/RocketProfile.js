// import React from 'react';
import { useSelector } from 'react-redux';

const RocketProfile = () => {
  const rockets = useSelector((state) => state.rocketsReducer.rockets);
  let bookedRockets;
  if (rockets) {
    bookedRockets = rockets.filter((rocket) => rocket.reserved);
  }
  return (
    <div>
      <h2>My Rockets</h2>
      {bookedRockets && (
        <ul>
          {bookedRockets.map((rocket) => (
            <li key={rocket.id}>{rocket.rocket_name}</li>
          ))}
        </ul>
      )}
      {(!bookedRockets || !bookedRockets[0]) && (
        <h4>You have not reserved any rocket!</h4>
      )}
    </div>
  );
};

export default RocketProfile;
