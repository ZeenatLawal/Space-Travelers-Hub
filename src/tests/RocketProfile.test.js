import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketProfile from '../components/RocketProfile';

it('renders correctly', () => {
  const profile = renderer
    .create(<Provider store={store}><RocketProfile /></Provider>)
    .toJSON();
  expect(profile).toMatchSnapshot();
});
