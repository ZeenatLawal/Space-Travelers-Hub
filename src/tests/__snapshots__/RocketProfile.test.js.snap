// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rocketProfileWrap"
>
  <h4
    className="myRockets"
  >
    My Rockets
  </h4>
  <h5
    className="noReservation"
  >
    No reservation yet!
  </h5>
</div>
`;
